rotkehlchen/history/typing.py

```diff
@@ -0,0 +1,53 @@
+from enum import Enum
+from typing import TYPE_CHECKING, NamedTuple, Union
+
+from rotkehlchen.errors import DeserializationError
+
+if TYPE_CHECKING:
+    from rotkehlchen.externalapis.coingecko import Coingecko
+    from rotkehlchen.externalapis.cryptocompare import Cryptocompare
+
+
+HistoricalPriceOracleInstance = Union['Coingecko', 'Cryptocompare']
+
+
+class HistoricalPriceOracleProperties(NamedTuple):
```

Comment:
I think this abstraction will be useful for having more granularity over the oracle calls later on. At least I hope so.